Make RPC console welcome message translation-friendly

hebasto opened the pull request:
The best practice is do not split a translatable multi-line message into single lines. This helps translators to follow the context.

jarolrod commented on src/qt/rpcconsole.cpp:
```diff
@@ -806,20 +806,25 @@ void RPCConsole::clear(bool keep_prompt)
+        tr("Welcome to the %1 RPC console.\n"
+           "Use up and down arrows to navigate history, and %2 to clear screen.\n"
+           "Type %3 for an overview of available commands.\n"
+           "For more information on using this console type %4.\n"
```
```suggestion
           "For more information on using this console, type %4.\n"
```

While here, we could improve the grammar by adding in a comma

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/331#issuecomment-841674393).

jarolrod commented on src/qt/rpcconsole.cpp:
```diff
@@ -806,20 +806,25 @@ void RPCConsole::clear(bool keep_prompt)
+        tr("Welcome to the %1 RPC console.\n"
+           "Use up and down arrows to navigate history, and %2 to clear screen.\n"
+           "Type %3 for an overview of available commands.\n"
+           "For more information on using this console type %4.\n"
```
```suggestion
           "For more information on using this console, type %4.\n"
```

While here, we could improve the grammar by adding in a comma

jarolrod commented on src/qt/rpcconsole.cpp:
```diff
@@ -806,20 +806,25 @@ void RPCConsole::clear(bool keep_prompt)
-                        "</span>"),
-                        true);
+    static const QString welcome_message =
+        /*: RPC console welcome message.
```
Any reason to make this multi-line? Could just be: 

```
//: RPC console welcome message. Placeholders %5 and %6 are style tags for the warning content
```

jarolrod replied:
Now that this message is longer, it's appropriate that it is multiline

jarolrod commented on src/qt/rpcconsole.cpp:
```diff
@@ -806,20 +806,25 @@ void RPCConsole::clear(bool keep_prompt)
-                        true);
+    static const QString welcome_message =
+        /*: RPC console welcome message.
+            Placeholders %5 and %6 are style tags for the warning content. */
```
Maybe we should expand upon this. args %1,%2,%3,%4 are written space-separated. This sets the precedent that these arguments are either followed or preceded with space. We should make it explicitly clear that %5 and %6 are not space separated from the rest of the text in this translator comment.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/331#issuecomment-841674393).

jarolrod reviewed (commented):
Concept ACK

hebasto commented:
Updated b00aaf6134daefd16a1ff5804a8cb45f6f223392 -> a596381c7244157d15063a1c590a993045ab5179 ([pr331.01](https://github.com/hebasto/gui/commits/pr331.01) -> [pr331.02](https://github.com/hebasto/gui/commits/pr331.02), [diff](https://github.com/hebasto/gui/compare/pr331.01..pr331.02)):

- addressed @jarolrod's comments

jarolrod reviewed (commented):
ACK a596381c7244157d15063a1c590a993045ab5179

hebasto commented:
Updated a596381c7244157d15063a1c590a993045ab5179 -> 67bef11e57f12ceacd3810008914c33b37c3c5bf ([pr331.02](https://github.com/hebasto/gui/commits/pr331.02) -> [pr331.03](https://github.com/hebasto/gui/commits/pr331.03), [diff](https://github.com/hebasto/gui/compare/pr331.02..pr331.03)):

- addressed @jarolrod's [comment](https://github.com/bitcoin-core/gui/pull/281#discussion_r635492736)

---

Seeing no reason to use `QString` optimization tricks as the `welcome_message` is `static` now.

jarolrod commented:
re-ACK 67bef11e57f12ceacd3810008914c33b37c3c5bf

hebasto commented:
Rebased 67bef11e57f12ceacd3810008914c33b37c3c5bf -> 0f3d955a38fe59a4e8cc4bf9f4442e6e1fd8bcbb ([pr331.03](https://github.com/hebasto/gui/commits/pr331.03) -> [pr331.04](https://github.com/hebasto/gui/commits/pr331.04)) due to the conflict with #281.

jarolrod commented:
re-ack 0f3d955a38fe59a4e8cc4bf9f4442e6e1fd8bcbb

hebasto commented:
@goums @Talkless Do you mind reviewing this PR?
